h/services/user_signup.py

```diff
@@ -113,18 +110,20 @@ def _require_activation(self, user):
         self.session.flush()
 
         # Send the activation email
-        mail_params = self.signup_email(
-            user_id=user.id, email=user.email, activation_code=user.activation.code
+        mail_params = signup.generate(
+            request=self.request,
+            user_id=user.id,
+            email=user.email,
+            activation_code=user.activation.code,
         )
-        self.mailer.send.delay(*mail_params)
+        tasks_mailer.send.delay(*mail_params)
```

Comment:
I think all of this injection happening here was intended to make testing easier, but I think it kind of did the reverse by:

 * Bloating the interface
 * Causing us to have to pass in weird things like partials which are hard to test and mock

This is the only way this is ever configured to run, and the extra degrees for freedom are just confusing and indirect.